test/test_fx.py

```diff
@@ -563,6 +563,165 @@ def test_construct_root_dict(self):
 
+    def test_split_module_into_submodules(self):
+        class MyModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.param = torch.nn.Parameter(torch.rand(3, 4))
+                self.linear = torch.nn.Linear(4, 5)
+
+            def forward(self, x, y):
+                z = self.linear(x + self.param).clamp(min=0.0, max=1.0)  # first subgraph target
+                w = self.linear(y).clamp(min=0.0, max=1.0)  # second subgraph target
+                return z + w
+
+        my_module = MyModule()
+        traced = torch.fx.symbolic_trace(my_module)
+
+        def make_module_have_submodules(
+            orig_graph_module: torch.fx.GraphModule, orig_torch_nn_module: torch.nn.Module
+        ):
+            # Creates a module from the nodes and root torch.nn.module of a sybmolically traced
+            # torch.fx.GraphModule
+            #
+            # TODO currently relies on idea that no node name changes are made between original graph and subgraph
+            # module_name: name for new module's root torch.nn.module
+            # module_node_names: list of node names to add to new graph, need to match names in orig_nodes
+            # orig_nodes: nodes from original torch.fx.GraphModule
+            # orig_base_module: original torch.nn.module which orig_nodes's torch.fx.GraphModule was traced from
+            # submodule_hints: if adding submodules from modules created by this function, add their names here
+            #                  and add their names to module_node_names and add their nodes to orig_nodes
+            def create_module_from_nodes(
+                module_name: str,
+                module_node_names,
+                orig_nodes: Dict[str, torch.fx.Node],
+                orig_base_module: torch.nn.Module,
+                submodule_hints=[]
+            ) -> (torch.fx.GraphModule, torch.fx.Node):
+                # create new graph
+                new_nodes: Dict[str, torch.fx.Node] = {}
+                graph = torch.fx.Graph()
+                graph_placeholders = (
+                    []
+                )  # used to keep track of placeholders to make torch.fx.Node out of created module
+
+                # copy over nodes from original graph
+                for node_name in module_node_names:
+                    orig_node = orig_nodes[node_name]
+
+                    # perform arg transformation
+                    new_args = []
+                    for orig_arg in orig_node.args:
+                        if isinstance(orig_arg, torch.fx.Node):
+                            # TODO assumption that nodes aren't nested in lists/tuples
+                            new_args.append(new_nodes[orig_arg.name])
+                        else:
+                            # TODO assumption that non-nodes are directly copyable
+                            new_args.append(orig_arg)
+                    new_args = tuple(new_args)
```

Comment:
Here and for kwargs you can use map_arg:
https://github.com/pytorch/pytorch/blob/29664e6aa3cd5b2fb9ae2dc27eecfc52f5ccd66f/torch/fx/graph.py#L55

That way it'll traverse through all the nested structures and replace `Node`s where appropriate